File: dashboard/dto/event.ts
type EventNames = 'person-enter' | 'person-leave';
type Severity = 'critical' | 'warning' | 'info';

interface Event {
    device: number[] | number
    data: any
}

interface ExtendedEvent extends Event {
    name: EventNames
    severity: Severity
}

export type EventDTO = Event | ExtendedEvent;

File: surveillance/dto/event.py
from typing_extensions import TypedDict, Literal, Any, List

# EventNames = Literal[Literal['person-enter'], Literal['person-leave']]
#
# Severity = Literal[Literal['critical'], Literal['warning'], Literal['info']]


class EventDTO(TypedDict):
    device: int
    data: Any


# class ExtendedEvent(Event, TypedDict):
#     name: EventNames
#     severity: Severity


# EventDTO = Event | ExtendedEvent
